backend/azure-functions/Controllers/ControllerFactory.ts

```diff
@@ -0,0 +1,41 @@
+import { Db, MongoClient } from "mongodb";
+import { ISettings } from "../Models/ISettings";
+import { ICollection } from "../Services/ICollection";
+import { EnvironmentSettings } from "../Models/EnvironmentSettings";
+import { TraceContext, HttpRequest } from "@azure/functions";
+import { ExampleController } from "./ExampleController";
+import { ExampleDataService } from "../Services/ExampleDataService";
+
+export class ControllerFactory {
```

Comment:
suggestion,
if i get it right, any new controller we are adding should be created from this factory,
now if we think of the steps to add new controller, it would be 2 steps,
add a create method here and add the actual class,
it can be reduced 1 step of just adding the new controller class, if we'll make the mongo more like a singleton provider.
it will also benefit us if we decouple a factory that aim to handle the creation of mongo collections from a factory of HTTP controllers

Reply:
i'm not sure to which object are you referring as a "singleton provider"

Reply:
the mongo db collection

Reply:
if i understand you correctly, you're suggesting removing the DataService class, and implementing its logic directly in the Controller's handlers (by passing the collection to the Controller directly and not to the DataService)?
If so, I don't think it's a good idea. It will make the handlers harder to read, and we wont be able to reuse that logic as easily. I think it makes sense to encapsulate the logic that handles DB operations in it's own class (the DataService) and not implement it in an HTTP controller.